tidy3d/plugins/microwave/impedance_calculator.py

```diff
@@ -82,14 +84,21 @@ def compute_impedance(self, em_field: MonitorDataTypes) -> IntegralResultTypes:
                 current = np.conj(2 * flux / voltage)
 
         impedance = voltage / current
+        impedance = ImpedanceCalculator._set_data_array_attributes(impedance)
         return impedance
 
     @pd.validator("current_integral", always=True)
     def check_voltage_or_current(cls, val, values):
         """Raise validation error if both ``voltage_integral`` and ``current_integral``
-        were not provided."""
+        are not provided."""
         if not values.get("voltage_integral") and not val:
             raise ValidationError(
-                "Atleast one of 'voltage_integral' or 'current_integral' must be provided."
+                "At least one of 'voltage_integral' or 'current_integral' must be provided."
             )
         return val
+
+    @staticmethod
+    def _set_data_array_attributes(data_array: IntegralResultTypes) -> IntegralResultTypes:
+        """Helper to set additional metadata for ``IntegralResultTypes``."""
+        data_array.name = "Z0"
+        return data_array.assign_attrs(units=OHM, long_name="characteristic impedance")
```

Comment:
is it possible to set this as `data_attrs` in `IntegralResultType`?

Reply:
if so, we could even just set the `data_array.name` without calling this helper.

Reply:
`IntegralResultTypes = Union[FreqDataArray, FreqModeDataArray, TimeDataArray]`

I wanted to avoid making unique classes for each possible output type, so I don't think that is possible. The name and attribute are simply for clarity when plotting or printing.